packages/context/src/context.ts

```diff
@@ -64,65 +112,170 @@ export class Context {
     return this.registry.has(key);
   }
 
+  /**
+   * Iterate all contexts by breadth first traversal of the graph following
+   * `parents`. For example, with the following context graph:
+   * - reqCtx -> [serverCtx, connectorCtx]
+   * - serverCtx -> [appCtx]
+   * - connectorCtx -> [appCtx]
+   * `reqCtx.getAllContexts()` returns an iterator of `[reqCtx, serverCtx,
+   * connectorCtx, appCtx]`.
+   *
+   * @param visitor A function to produce a result from the context object
+   * locally without consulting the parents.
+   */
+  protected *contextIterator<T>(
+    visitor: (ctx: Context) => {done: boolean; value: T},
+  ): IterableIterator<T> {
+    const sorted: Set<Context> = new Set();
```

Comment:
A good variable name describes describes the intent (check which nodes we have visited) rather than implementation details (use a sorted data structure providing fast look-up).

Proposed name: `visited`.